service/controller/v14/templates/cloudformation/guest/auto_scaling_group.go

```diff
@@ -1,31 +1,32 @@
 package guest
 
 const AutoScalingGroup = `{{define "autoscaling_group"}}
-  {{ .ASGType }}AutoScalingGroup:
+{{- $v := .Guest.AutoScalingGroup }}
```

Comment:
Added this variable so we don't have to repeat this stuff over and over and make noise in the templates.

Reply:
Not a fan of the templates variables. This makes it just worse the more complex the input types get. Having this explicit would make total sense to me. Below you have still `$v.Master.DockerVolume.ResourceName` and you wonder what `$v` actually is.

Reply:
In this case used carefully they reduce unnecessary noise and make template reading easier. They also won't get worse as input gets more complex, because all those variables follow the same pattern `{{- $v := .CF_STACK_NAME.SUB_TEMPLATE_NAME }}`. I agree abuse of template variables can lead to very complex, unreadable templates, but this definitely isn't the case. I'd like to keep them.